src/main/java/org/embulk/input/aws_cost_explorer/client/AwsCostExplorerResponse.java

```diff
@@ -0,0 +1,112 @@
+package org.embulk.input.aws_cost_explorer.client;
+
+import com.amazonaws.services.costexplorer.model.GetCostAndUsageResult;
+import com.amazonaws.services.costexplorer.model.MetricValue;
+import com.amazonaws.services.costexplorer.model.ResultByTime;
+import org.embulk.input.aws_cost_explorer.AwsCostExplorerInputPlugin.PluginTask;
+import org.embulk.spi.PageBuilder;
+import org.embulk.spi.Schema;
+import org.embulk.spi.time.Timestamp;
+import org.embulk.spi.time.TimestampParser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AwsCostExplorerResponse
+{
+    private final GetCostAndUsageResult result;
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private final TimestampParser timestampParser = TimestampParser.of("%Y-%m-%d", "UTC");
+
+    /**
+     * Constructor
+     *
+     * @param result GetCostAndUsageResult
+     */
+    public AwsCostExplorerResponse(GetCostAndUsageResult result)
+    {
+        this.result = result;
+    }
+
+    /**
+     * Get next page token
+     */
+    public String getNextPageToken()
+    {
+        return result.getNextPageToken();
+    }
+
+    /**
+     * Add records to page
+     *
+     * @param pageBuilder PageBuilder
+     * @param task PluginTask
+     */
+    public void addRecordsToPage(PageBuilder pageBuilder, PluginTask task)
+    {
+        result.getResultsByTime().forEach(resultsByTime -> {
+            System.out.println(resultsByTime.toString());
+            logger.info("Cost Explorer API results: {}", resultsByTime);
+
+            if (resultsByTime.getGroups().isEmpty()) {
+                addRecordToPageWithoutGroupBy(pageBuilder, task, resultsByTime);
+            }
+            else {
+                addRecordsToPageWithGroupBy(pageBuilder, task, resultsByTime);
+            }
+        });
+    }
+
+    private void addRecordToPageWithoutGroupBy(PageBuilder pageBuilder, PluginTask task, ResultByTime resultsByTime)
```

Comment:
nits: There is a duplication in addRecordsToPageWithGroupBy, which may need to be refactored if there is time.